### ISSUE TITLE
SDWebImagePrefetcher consistently skips chunks of images

### ISSUE THREAD

**Issue by IljaDaderko** [SDWebImage/SDWebImage#2732]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 12
 SDWebImage Version      | 5.0.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 10
 Repro rate              | all the time (100%)

### Issue Description and Steps

I am passing array of webp image urls to prefetched with completion block in my case it always returns roughly 50/50 of finished images and skipped images i.e. 80 finished 70 skipped

```objc
@try {
    [[SDWebImagePrefetcher sharedImagePrefetcher]
        prefetchURLs:urls
            progress:nil
           completed:^(NSUInteger noOfFinishedUrls, NSUInteger noOfSkippedUrls) {
             NSNumber *finished = [NSNumber numberWithLong:noOfFinishedUrls];
             NSNumber *skipped = [NSNumber numberWithLong:noOfSkippedUrls];
             // ...
           }];
  } @catch (NSError *error) {
    reject(@"ASQImage:unknown_error", @"prefetch method failed", error);
  }
```

There is no error thrown and I'm not sure why this is happening. Any suggestions on how to debug this perhaps?

**Comment by IljaDaderko** [comment]
After debugging this for a bit I can see following.

1. I logged `urls` that I am passing to prefetcher, all of them are there (all 80 urls that look like this `http://localhost:8081/assets/packages/skimitar-assets/images/elf_background.jpg?platform=ios&hash=8bafc50df42fc90206453f1f654d0b12`) So I am confident I am passing correct data to prefetcher

2. I added a log here https://github.com/SDWebImage/SDWebImage/blob/master/SDWebImage/SDWebImagePrefetcher.m#L126 to log out error related to skipped images, and the reason why they are skipped is that "Image url was nil"

I don't understand why it is nil if I can see that I am passing all urls to it?
